warehouse/redshift.go

```diff
@@ -135,22 +135,27 @@ func (rs *Redshift) LoadToWarehouse(file string) error {
 		log.Printf("Failed to upload file %s to s3: %s", file, err)
 		return err
 	}
-	defer rs.RemoveS3Object(s3obj)
 
-	rs.conn, err = rs.MakeRedshfitConnection()
-	if err != nil {
-		return err
-	}
-	defer rs.conn.Close()
+	if rs.conf.S3.S3Only {
+		log.Printf("Config flag S3Only is on, skipping load to Redshift")
+	} else {
+		defer rs.RemoveS3Object(s3obj)
```

Comment:
Rather than put the redshift logic in an `else` clause, it would be cleaner to return from the `if rs.conf.S3.S3Only` clause. Also, rather than log `skipping load to redshift` on each file, I'd just do it once on startup (since the flag isn't dynamic).